lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -1519,6 +1519,18 @@ static bool checkSingleOverride(ValueDecl *override, ValueDecl *base) {
     diagnoseOverrideForAvailability(override, base);
   }
 
+  // Overrides of NSObject.hashValue are deprecated; one should override
+  // NSObject.hash instead.
+  if (auto baseVar = dyn_cast<VarDecl>(base)) {
+    if (auto classDecl =
+          baseVar->getDeclContext()->getAsClassOrClassExtensionContext()) {
+      if (classDecl->getBaseName().userFacingName() == "NSObject" &&
```

Comment:
This will also trigger on custom classes that happen to be named `NSObject`, right? Would it be possible to make this a more specific match?

Reply:
Sure, I can check for the `ObjectiveC` module.

Reply:
... or `Foundation`, I guess, for corelibs Foundation.

Reply:
Oof, that's fun. Good catch.